pkgs/tools/networking/dae/default.nix

```diff
@@ -0,0 +1,47 @@
+{ lib
+, llvmPackages_15
+, fetchFromGitHub
+, buildGoModule
+}:
+buildGoModule rec {
+  pname = "dae";
+  version = "0.1.5";
+
+  src = fetchFromGitHub {
+    owner = "daeuniverse";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "sha256-EoStRmyYOtvG5ejtvHWNe9IIeE77hqp1OXBhRdxCYHs=";
+    fetchSubmodules = true;
+  };
+
+  vendorHash = "sha256-vxHufE3538l6zIcozFcrNhl+2sG1PtzkVxC0NyL3WMU=";
+
+  proxyVendor = true;
+
+  GOFLAGS = "-buildvcs=false";
+
+  ldflags = [
+    "-s"
+    "-w"
+    "-X github.com/daeuniverse/dae/cmd.Version=${version}"
+    "-X github.com/daeuniverse/dae/common/consts.MaxMatchSetLen_=64"
+  ];
+
+  preBuild = ''
+    unset STRIP
+    make CFLAGS="-D__REMOVE_BPF_PRINTK -fno-stack-protector" \
```

Comment:
What are these `CFLAGS` for?

Reply:
> What are these `CFLAGS` for?

Flag `-D__REMOVE_BPF_PRINTK` is for preventing [Unknown bpf_trace_printk error](https://github.com/daeuniverse/dae/blob/85343ac141768d4bcdc48a2ba9edff56b3b9c1a1/docs/getting-started/troubleshooting.md?plain=1#L11).

Compiler in NixOS force enabled stack protector which will causes compile fail. `-fno-stack-protector` for disabling it.

<details><summary>Log</summary>

```console
> nix build .#dae
error: builder for '/nix/store/qv3fxz1wv3mdxdf2zvzx0k2bba1kd6ig-dae-0.1.5.drv' failed with exit code 2;
       last 10 log lines:
       >     ^
       > /build/source/control/kern/tproxy.c:2173:5: error: A call to built-in function '__stack_chk_fail' is not supported.
       > int tproxy_wan_cg_sendmsg6(struct bpf_sock_addr *ctx) {
       >     ^
       > 11 errors generated.
       > Error: can't execute /nix/store/fam63d3x0fv1mb9i7igqikqzk1ngmcdx-clang-wrapper-11.1.0/bin/clang: exit status 1
       > exit status 1
       > control/control.go:8: running "go": exit status 1
       > make: *** [Makefile:51: ebpf] Error 1
       > /nix/store/sw36plhp82916wwg6i6097rkzza7d950-stdenv-linux/setup: line 136: pop_var_context: head of shell_variables not a function context
       For full logs, run 'nix log /nix/store/qv3fxz1wv3mdxdf2zvzx0k2bba1kd6ig-dae-0.1.5.drv'.
```
</details>